src/Okta.PowerShell/Private/OktaApiClient.ps1

```diff
@@ -163,7 +164,8 @@ function Invoke-OktaApiClient {
                                         -SkipCertificateCheck `
                                         -Proxy $Configuration["Proxy"].GetProxy($UriBuilder.Uri) `
                                         -ProxyUseDefaultCredentials `
-                                        -UserAgent $OktaUserAgent
+                                        -UserAgent $OktaUserAgent `
```

Comment:
Considering your comment about `IWR` not reaching the Retry logic due to `ErrorAction = Stop`, I'm curious how tests still passed. I left a question for you to help me reproduce the issue.

Reply:
I gave that a look and the tests pass because the calls to IWR are being Mock'ed out. For example in this block

https://github.com/okta/okta-powershell-cli/blob/f78b2c65092583c3477fa7273c67ee31a941368b/tests/Internal/OktaApiClient.Tests.ps1#L9-L27

Calls to `Get-OktaConfiguration` and `Invoke-WebRequest` are replaced by the script blocks in the Mock definition. When `Invoke-OktaListApplications` is called the inner calls to the two mocked functions run the script blocks not the actual commands so `Invoke-WebRequest` never gets called or a non-OK status response for IWR to throw an error

I cant think of an "easy" way to unit test IWR getting an actual 429 response thats not overly contrived - eg mock an internal part of that method - so powershells IWR method gets the 429, or call an actual weburl tat forcibly returns 429 - both seem more like integration tests than unit tests though